docs/DeveloperGuide.md

```diff
@@ -247,7 +247,43 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 _{more aspects and alternatives to be added}_
 
-
+### Command history feature
+#### Implementation
+The command history feature is facilitated by `CommandTextHistory`. It has a `commandHistory` arraylist that stores the history of commands entered by the user as strings 
+and a `currentCommandIndex` that is initialised to be `-1` to keep track of the current command index in the command history.
+The `CommandTextHistory` class implements the following operations:
+* `CommandTextHistory#addCommandHistory(String commandText)` — Adds a command text to the command history.
+* `CommandTextHistory#getPreviousCommand()` — Returns the previous command text in the command history.
+* `CommandTextHistory#getNextCommand()` — Returns the next command text in the command history.
+
+##### Getting previous/next command
+The `getPreviousCommand()` method will first check if `commandHistory` is empty by checking if `currentCommandIndex` is more than `0`. 
+If it is not empty, it will decrement the `currentCommandIndex` and return the command text at that index.
+If it is empty, it will return an empty string.
+
+The `getNextCommand()` method will first check if `commandHistory` if `currentCommandIndex` is less than `commandHistory.size() - 1`.
+If it is, it will increment the `currentCommandIndex` and return the command text at that index.
+If it is not or `currentCommandIndex` is `-1` which means the commandHistory is empty, it will return an empty string.
```

Comment:
I think when it's at the end of the `commandHistory`, the `currentCommandIndex` is the index of the last command (not sure why it would be `-1`)? But agree that we could probably include the case where it doesn't increment the `currentCommandIndex` but still returns the command text when it's at the end.

Reply:
I think its because currently the implementation just sets the currentCommandIndex to -1 on initialisation.

![image](https://github.com/user-attachments/assets/cdd9b280-814f-441a-ad3f-f728c95f12f0)

Reply:
Okk I see, but should we include the case where the the index refers to the last command in the list, where no next command is available to be returned?